### ISSUE TITLE
[Feature Request] Allow sites to have their own Azure Search Index assigned to them

### ISSUE THREAD

**Issue by ddavisfw** [PiranhaCMS/piranha.core#1765]
Currently I have customized this in my solution to allow this to work. I have a site that is set up to handle business "events" that occur quarterly. A one-size-fits-all solution for search indexes does not work for this. Deleting an event (site) should delete the azure search index for our needs as we are limited to X indexes per search service. Having the ability to create an index for each new site and the option to remove it when the site is deleted would be helpful. It would be nice to get to the point where none of our changes are to the Piranha project itself and we can just inject everything into it from our web project. This feature is the only one preventing us from doing that.

**Comment by tidyui** [comment]
Just so I’m following you correctly, have you created a service implementating the Piranha ISearch interface, but you would like the interface to include an additional method triggered when a site is deleted?

Just want to make sure I’m not making false assumptions here!

**Comment by ddavisfw** [comment]
Yes and also the ability to pass in a specific index name when creating an index.

**Comment by ddavisfw** [comment]
I think there would need to be some association between the site and the index name in the database, maybe in the piraha_sites table.

**Comment by tidyui** [comment]
Given that search is optional we can’t introduce mandatory keys for a search engine. Can’t the site Id be used to create a unique index.

I think you need to specify more what your problem is cause I’m still not really following you!

Regards

**Comment by tidyui** [comment]
So, I started to look at this. For the unique site-index, when the method `ISearch.SavePageAsync` and `ISearch.DeletePageAsync` is called the model passed in will have the property `SiteId`. Isn't this property enough to use when generating a search-index per site?
